source/guide/3.2/guide-security.md

```diff
@@ -44,5 +44,54 @@ Availbale in the Commercial version only
 
 # Writing your own userstore and authentication providers
 ## how to write
-## how to pack
-## how to configure installation on Cloudify Manager
+
+## Packaging/Configuring/Installing custom implementations
+
+In order to use custom implementations of userstores, authentication providers and token generators; the implementations themselves should be installed on the manager.
+
+To do this, the manager blueprint should be updated as follows.
+
+Say you write a custom authentication provider. The code itself should be packaged in a similar way to how [operations/workflows](guide-plugin-creation.html#creating-a-plugin-project) plugins are packaged, that is to say, it should be packaged as a valid python package.
+
+You specify the package location under the `cloudify.plugins` property in the `manager` node of the manager blueprint like this:
+{% highlight yaml %}
+node_templates:
+  ...
+  manager:
+    ...
+    properties:
+      ...
+      cloudify:
+        plugins:
+          my_authentication_provider:
+
+            # see description below
+            source: my-extensions/simple-authentication-provider
+
+            # see description below
+            install_args: '--pre'
+
+          my_userstore:
+
+            # see description below
+            source: https://github.com/my-org/my-auth-provider/archive/master.zip
+
```

Comment:
i think a clarification (perhaps in a note) is needed that explains the difference between security plugins and regular plugin. so that people wont suddenly start writing operations or workflows inside these plugin with the thought that the can later reference them.